mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb

```diff
@@ -5145,7 +5160,7 @@ func expandResourceUsageExportConfig(configured interface{}) *container.Resource
 func expandDnsConfig(configured interface{}) *container.DNSConfig {
 	l := configured.([]interface{})
 	if len(l) == 0 || l[0] == nil {
-		return nil
+		return &container.DNSConfig{}
```

Comment:
It looks like this change is requiring a large number of our test recordings to fail in replaying mode. Is it necessary in order to enable update support for this field?

Reply:
This is to allow returning to default DNSConfig by removing it completely. Otherwise, the API call would have `DesiredDNSConfig: nil` which does not change it.

So not providing the DNSConfig in TF should have DesiredDNSConfig as an empty object rather than nil so that it is updated to the default DNSConfig.

Reply:
This seems like it could be a change to existing resources. Would the API behavior change for resources that do not specify `dns_config`?

Reply:
It will ideally affect others that don't specify DNSConfig but do not have the default one. Anyway I reverted the change to move on with the PR. Since it concerns disabling where the PR main goal is in-place enabling.